functions/spanner/src/main/java/functions/HelloSpanner.java

```diff
@@ -0,0 +1,123 @@
+import com.google.cloud.functions.HttpRequest;
+import com.google.cloud.functions.HttpResponse;
+import com.google.cloud.spanner.DatabaseClient;
+import com.google.cloud.spanner.DatabaseId;
+import com.google.cloud.spanner.ResultSet;
+import com.google.cloud.spanner.Spanner;
+import com.google.cloud.spanner.SpannerException;
+import com.google.cloud.spanner.SpannerOptions;
+import com.google.cloud.spanner.Statement;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.MoreObjects;
+import java.io.BufferedWriter;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+// HelloSpanner is an example of querying Spanner from a Cloud Function.
+public class HelloSpanner implements HttpFunction {
+  private static final Logger LOGGER = Logger.getLogger(HelloSpanner.class.getName());
+
+  @VisibleForTesting
+  static Spanner createSpanner() {
+    return SpannerOptions.newBuilder().build().getService();
+  }
+
+  // SpannerHolder is a holder class for a Spanner instance that is initialized lazily.
+  private static final class SpannerHolder {
+    private final Object lock = new Object();
+    private volatile boolean initialized;
+    private volatile DatabaseClient client;
+    private volatile Throwable error;
+
+    private SpannerHolder() {}
+
+    // Initialize the {@link Spanner} instance in a method and not as a static variable, as it
+    // might throw an error, and we want to catch and log that specific error.
+    DatabaseClient get() throws Throwable {
+      if (!initialized) {
+        synchronized (lock) {
+          if (!initialized) {
+            try {
+              DatabaseId db =
+                  DatabaseId.of(
+                      SpannerOptions.getDefaultProjectId(),
+                      SPANNER_INSTANCE_ID,
+                      SPANNER_DATABASE_ID);
+              client = createSpanner().getDatabaseClient(db);
+            } catch (Throwable t) {
+              error = t;
+            }
+            initialized = true;
+          }
+        }
+      }
+      if (error != null) {
+        throw error;
+      }
+      return client;
+    }
+  }
+
+  // The SpannerHolder instance is shared across all instances of the HelloSpanner class.
+  private static final SpannerHolder SPANNER_HOLDER = new SpannerHolder();
+
+  @VisibleForTesting
+  DatabaseClient getClient() throws Throwable {
+    return SPANNER_HOLDER.get();
+  }
+
+  // TODO<developer>: Set these environment variables.
+  private static final String SPANNER_INSTANCE_ID =
+      MoreObjects.firstNonNull(System.getenv("SPANNER_INSTANCE"), "my-instance");
+  private static final String SPANNER_DATABASE_ID =
+      MoreObjects.firstNonNull(System.getenv("SPANNER_DATABASE"), "example-db");
+
+  @Override
+  public void service(HttpRequest request, HttpResponse response) throws Exception {
+    BufferedWriter w = response.getWriter();
+    try {
+      DatabaseClient client = getClient();
+      try (ResultSet rs =
```

Comment:
Nit: would `results` be clearer?

Reply:
`rs` is a very common abbreviation for `ResultSet`. See for example https://docs.oracle.com/javase/tutorial/jdbc/basics/processingsqlstatements.html.